### DISCUSSION TITLE
Add before method to the migrations

### DISCUSSION THREAD

**Discussion by elisonkaloshi** [laravel/framework#48887] [Ideas]
It is possible to add the `before` method to the migrations in Laravel
For example:
I have a table with 10 columns and it will be easier and faster to just use before('created_at'), rather search who is the last column before the created_at column.

`$table->string('description')->after('client_name')->nullable();`

`$table->string('description')->before('created_at')->nullable();`

**Comment by shaedrich** [comment]
It's funny but I always just assumed, there was already one ^^ Checked the docs and the code and nope, you're right. Weird. Yeah, would absolutely make sense to have it.

**Comment by hafezdivandari** [comment]
That is because MySQL doesn't support this, it only supports `AFTER` and `FIRST`.

> To add a column at a specific position within a table row, use FIRST or AFTER col_name. The default is to add the column last. [Docs]( https://dev.mysql.com/doc/refman/8.0/en/alter-table.html#:~:text=To%20add%20a%20column%20at%20a%20specific%20position%20within%20a%20table%20row%2C%20use%20FIRST%20or%20AFTER%20col_name.%20The%20default%20is%20to%20add%20the%20column%20last.)

**Reply by shaedrich** [reply]
It could still be implemented as syntactic sugar for what @elisonkaloshi is now doing manually. The timing couldn't be any better since Laravel just deprecated doctrine and implemented native methods for doctrine functionality.

**Reply by hafezdivandari** [reply]
You are right, one may send a PR using `Schema::getColumnListing` to implement this.
